libraries/AP_InertialSensor/AP_InertialSensor_SCHA63T.cpp

```diff
@@ -0,0 +1,426 @@
+
+    return sensor;
+}
+
+void AP_InertialSensor_SCHA63T::start()
+{
+    if (!_imu.register_accel(accel_instance, ACCEL_BACKEND_SAMPLE_RATE, dev_accel->get_bus_id_devtype(DEVTYPE_INS_SCHA63T)) ||
+        !_imu.register_gyro(gyro_instance, GYRO_BACKEND_SAMPLE_RATE,   dev_gyro->get_bus_id_devtype(DEVTYPE_INS_SCHA63T))) {
+        return;
+    }
+
+    // setup sensor rotations from probe()
+    set_gyro_orientation(gyro_instance, rotation);
+    set_accel_orientation(accel_instance, rotation);
+
+    // setup callbacks
+    dev_accel->register_periodic_callback(1000000UL / ACCEL_BACKEND_SAMPLE_RATE,
+                                          FUNCTOR_BIND_MEMBER(&AP_InertialSensor_SCHA63T::read_fifo_accel, void));
+    dev_gyro->register_periodic_callback(1000000UL / GYRO_BACKEND_SAMPLE_RATE,
+                                         FUNCTOR_BIND_MEMBER(&AP_InertialSensor_SCHA63T::read_fifo_gyro, void));
+}
+
+/*
+  probe and initialise accelerometer
+ */
+bool AP_InertialSensor_SCHA63T::init()
+{
+    WITH_SEMAPHORE(dev_accel->get_semaphore());
+    WITH_SEMAPHORE(dev_gyro->get_semaphore());
+
+    uint8_t v[4];
+
+    hal.scheduler->delay(25);
+
+    RegisterWrite(0, SEL_BANK, 0);
+    RegisterWrite(0, RESCTRL, 0);
+    hal.scheduler->delay(1);
+    RegisterWrite(1, SEL_BANK, 0);
+    RegisterWrite(1, RESCTRL, 0);
+
+    hal.scheduler->delay(50);
+
+    RegisterWrite(0, MODE, 0);
+    RegisterWrite(0, G_FILT_DYN, 0);
+    RegisterWrite(0, A_FILT_DYN, 0);
+    hal.scheduler->delay(1);
+    RegisterWrite(1, MODE, 0);
+    RegisterWrite(1, MODE, 0);
+    RegisterWrite(1, G_FILT_DYN, 0);
+
+    hal.scheduler->delay(500);
+
+    RegisterWrite(0, SET_EOI, 0);
+    RegisterWrite(1, SET_EOI, 0);
+
+    hal.scheduler->delay(50);
+
+    RegisterRead(0, S_SUM, v);
+    RegisterRead(1, S_SUM, v);
+    hal.scheduler->delay(50);
+    RegisterRead(0, S_SUM, v);
+    RegisterRead(1, S_SUM, v);
+    hal.scheduler->delay(50);
+    RegisterRead(0, S_SUM, v);
+    RegisterRead(1, S_SUM, v);
+    hal.scheduler->delay(50);
+    int ret_uno = RegisterRead(0, S_SUM, v);
+    int ret_due = RegisterRead(1, S_SUM, v);
+
+    return ret_uno && ret_due;
+}
+
+/*
+  read accel fifo
+ */
+void AP_InertialSensor_SCHA63T::read_fifo_accel(void)
+{
+    static const uint8_t cmd_accl_x[4] = { 0x10, 0x00, 0x00, 0xE9 }; // ACCL_X
+    static const uint8_t cmd_accl_y[4] = { 0x14, 0x00, 0x00, 0xEF }; // ACCL_Y
+    static const uint8_t cmd_accl_z[4] = { 0x18, 0x00, 0x00, 0xE5 }; // ACCL_Z
+    static const uint8_t cmd_rate_x[4] = { 0x04, 0x00, 0x00, 0xF7 }; // RATE_X
+    static const uint8_t cmd_temper[4] = { 0x1C, 0x00, 0x00, 0xE3 }; // TEMPER
+
+    uint8_t rsp_accl_x[4];
+    uint8_t rsp_accl_y[4];
+    uint8_t rsp_accl_z[4];
+    uint8_t rsp_rate_x[4];
+    uint8_t rsp_temper[4];
+
+    int16_t accel_x;
+    int16_t accel_y;
+    int16_t accel_z;
+
+    // #### ACCL_X Cmd Send (This Response rsp_accl_x is Dust!!)
+    dev_accel->transfer_fullduplex(cmd_accl_x, rsp_accl_x, 4);
```

Comment:
you can just call transfer() with both send_len and recv_len set as 4

Reply:
Unlike transfer(), this sensor uses transfer_fullduplex() because transmission and reception are performed at the same time.